### PR TITLE
Codepush initial structure and enums

### PR DESCRIPTION
**Pull request by annakocheshkova**
Added codepush project structure and enums.

### CODE REVIEW

**Comment by sergey-akhalkov** [comment]
Also, we've added fake tests to pass `VSTS: AppCenter-SDK-Android Build and Test` check. We'll replace fake tests with real in the future, please let us know if you have any concerns or considerations.

**Comment by sergey-akhalkov** [comment]
Ahh I got it, we neet to have 100% code coverage - we’ll do for this and further PRs.

**Comment by guperrot** [comment]
@sergey-akhalkov if that was only code coverage you would have green build but red codecov status. There is another issue with the `gradlew asemble lint` step. Please run that too and check.

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/enums/CodePushUpdateState.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+public enum CodePushUpdateState {
+
+    /**
+     * Indicates that an update represents the
+     * version of the app that is currently running.
+     */
+    @SerializedName("0")
+    RUNNING(0),
+
+     /**
```
Comment (guperrot) [inline sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/enums/CodePushUpdateState.java]: please format file again (with defaults), I see an extra space here.

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/enums/CodePushUpdateState.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+    @SerializedName("2")
+    LATEST(2);
+
+    private final int value;
+    
+    CodePushUpdateState(int value) {
+        this.value = value;
+    }
+
+    public int getValue() {
```
Comment (guperrot) [inline sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/enums/CodePushUpdateState.java]: public/protected methods need javadoc

File: sdk/appcenter-codepush/src/test/java/com/microsoft/appcenter/codepush/CodePushTest.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package com.microsoft.appcenter.codepush;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class CodePushTest {
+    @Test
```
Comment (guperrot) [inline sdk/appcenter-codepush/src/test/java/com/microsoft/appcenter/codepush/CodePushTest.java]: please add blank line before annotations such as @Test or @Override, also no underscore in method names, use camelCase

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/enums/CodePushSyncStatus.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    @SerializedName("8")
+    INSTALLING_UPDATE(8);
+
+    private final int value;
+
+    CodePushSyncStatus(int value) {
+        this.value = value;
+    }
+    
+    public int getValue() {
```
Comment (guperrot) [inline sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/enums/CodePushSyncStatus.java]: Javadoc missing here

File: sdk/appcenter-codepush/src/androidTest/java/com/microsoft/appcenter/codepush/CodePushTest.java [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package com.microsoft.appcenter.codepush;
+
+import android.support.test.runner.AndroidJUnit4;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.*;
+
+@RunWith(AndroidJUnit4.class)
```
Comment (guperrot) [inline sdk/appcenter-codepush/src/androidTest/java/com/microsoft/appcenter/codepush/CodePushTest.java]: Remove this annotation, its already junit 4 by default in all our tests

File: sdk/appcenter-codepush/src/androidTest/java/com/microsoft/appcenter/codepush/CodePushTest.java [hunk 6]
```diff
@@ -0,0 +1,17 @@
+package com.microsoft.appcenter.codepush;
+
+import android.support.test.runner.AndroidJUnit4;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import static org.junit.Assert.*;
```
Comment (guperrot) [inline sdk/appcenter-codepush/src/androidTest/java/com/microsoft/appcenter/codepush/CodePushTest.java]: never use star imports, please configure your commit dialog to cleanup imports and format code etc... or do it manually with IDE actions

File: sdk/appcenter-codepush/build.gradle [hunk 7]
```diff
@@ -0,0 +1,7 @@
+project.description = 'A cloud service that enables developers to deploy mobile app updates directly to their users` devices.'
+evaluationDependsOn(':sdk')
+
+dependencies {
+    compile project(':sdk:appcenter')
+    compile "com.google.code.gson:gson:2.8.2"
```
Comment (jaelim-ms) [inline sdk/appcenter-codepush/build.gradle]: Is it necessary?

File: sdk/appcenter-codepush/src/androidTest/java/com/microsoft/appcenter/codepush/CodePushTest.java (hunk 6 above)
Comment (annakocheshkova) [reply sdk/appcenter-codepush/src/androidTest/java/com/microsoft/appcenter/codepush/CodePushTest.java]: Fixed it, thanks.

File: sdk/appcenter-codepush/src/androidTest/java/com/microsoft/appcenter/codepush/CodePushTest.java (hunk 5 above)
Comment (annakocheshkova) [reply sdk/appcenter-codepush/src/androidTest/java/com/microsoft/appcenter/codepush/CodePushTest.java]: @guperrot Removed it, thanks.

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/enums/CodePushSyncStatus.java (hunk 4 above)
Comment (annakocheshkova) [reply sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/enums/CodePushSyncStatus.java]: @guperrot Added javadoc, thanks.

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/enums/CodePushUpdateState.java (hunk 1 above)
Comment (annakocheshkova) [reply sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/enums/CodePushUpdateState.java]: @guperrot Formatted file, thanks.

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/enums/CodePushUpdateState.java (hunk 2 above)
Comment (annakocheshkova) [reply sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/enums/CodePushUpdateState.java]: @guperrot Added javadoc, thanks.

File: sdk/appcenter-codepush/src/test/java/com/microsoft/appcenter/codepush/CodePushTest.java (hunk 3 above)
Comment (annakocheshkova) [reply sdk/appcenter-codepush/src/test/java/com/microsoft/appcenter/codepush/CodePushTest.java]: @guperrot Reformatted, thanks.

File: sdk/appcenter-codepush/build.gradle (hunk 7 above)
Comment (annakocheshkova) [reply sdk/appcenter-codepush/build.gradle]: @jaelim-ms Yes, we have discussed it recently with the members of the team and decided that it's fine to use gson library. We have a lot of data classes in our project and it is more comfortable for us to use this library than serializing/de serializing manually.

File: sdk/appcenter-codepush/src/androidTest/java/com/microsoft/appcenter/codepush/CodePushAndroidTest.java [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class CodePushAndroidTest {
+
+    @Test
+    public void useAppContext() throws Exception {
+        
+        // Context of the app under test.
```
Comment (guperrot) [inline sdk/appcenter-codepush/src/androidTest/java/com/microsoft/appcenter/codepush/CodePushAndroidTest.java]: All our comments use `/*  */` except the ones generated by Android Studio do disable a warning.

Also every blank line in a method must be followed by a new comment to start the new block.

File: sdk/appcenter-codepush/build.gradle (hunk 7 above)
Comment (guperrot) [inline sdk/appcenter-codepush/build.gradle]: Please use single quotes for dependencies when there is no version variable.

File: sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/enums/CodePushCheckFrequency.java [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    ON_APP_START(0),
+
+    /**
+     * When the app re-enters the foreground.
+     */
+    @SerializedName("1")
+    ON_APP_RESUME(1),
+
+    /**
+     * Don't automatically check for updates, but only do it when <code>codePush.sync()</code> is manully called inside app code.
```
Comment (dhei) [inline sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/enums/CodePushCheckFrequency.java]: typo: "manually"

Comment (annakocheshkova) [reply sdk/appcenter-codepush/src/main/java/com/microsoft/appcenter/codepush/enums/CodePushCheckFrequency.java]: @dhei Fixed, thanks.

File: sdk/appcenter-codepush/build.gradle (hunk 7 above)
Comment (annakocheshkova) [reply sdk/appcenter-codepush/build.gradle]: @guperrot Got it, thanks. Fixed.

File: sdk/appcenter-codepush/src/androidTest/java/com/microsoft/appcenter/codepush/CodePushAndroidTest.java (hunk 8 above)
Comment (annakocheshkova) [reply sdk/appcenter-codepush/src/androidTest/java/com/microsoft/appcenter/codepush/CodePushAndroidTest.java]: @guperrot Fixed

**Comment by guperrot** [comment]
Looks good now, we need to have discussion with @dougnutz before we can merge anything though.

**Comment by guperrot** [comment]
Since https://msmobilecenter.visualstudio.com/Mobile-Center/_workitems/edit/31611 has been cancelled, let's close this PR for now and reopen later when this work will be active again.

**Pull request closed without merging**